packages/block-library/src/navigation/edit/index.js

```diff
@@ -279,14 +279,24 @@ function Navigation( {
 		setIsPlaceholderShown( ! isEntityAvailable );
 	}, [ isEntityAvailable ] );
 
+	// If the ref no longer exists the reset the inner blocks
+	// to provide a clean slate.
+	useEffect( () => {
+		if ( ref === undefined && innerBlocks.length > 0 ) {
+			replaceInnerBlocks( clientId, [] );
+		}
+		// innerBlocks are intentionally not listed as deps. This function is only concerned
+		// with the snapshot from the time when ref became undefined.
+	}, [ clientId, ref ] );
```

Comment:
```suggestion
	}, [ clientId, ref, innerBlocks ] );
```

Do we now need to add `innerBlocks` to the deps?

Reply:
No, as the comment above states :-)

Reply:
Oops, looks like it was added anyway in 793ff2a

Reply:
I don't even know what happened here 😕